patches/api/0417-Add-PlayerShieldDisableEvent.patch

```diff
@@ -0,0 +1,55 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cryptite <cryptite@gmail.com>
+Date: Mon, 1 May 2023 16:22:43 -0500
+Subject: [PATCH] Add PlayerShieldDisableEvent
+
+Called whenever a player's shield is disable. The cooldown (default 100 ticks) is also provided to optionally be changed
+
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerShieldDisableEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerShieldDisableEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..312c753773be6fa11ad640bab1842502947c2dc2
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerShieldDisableEvent.java
+@@ -0,0 +1,42 @@
++package io.papermc.paper.event.player;
++
++import org.bukkit.Material;
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Thrown whenever a Player's shield is disabled
++ */
++public class PlayerShieldDisableEvent extends PlayerItemCooldownEvent implements Cancellable {
++    private static final HandlerList HANDLER_LIST = new HandlerList();
```

Comment:
My idea actually involved not having a handler list on this event. That way a listener for PlayerItemCooldownEvent would also catch this one.